DuckDuckGo/AutofillLoginListViewModel.swift

```diff
@@ -195,12 +209,34 @@ final class AutofillLoginListViewModel: ObservableObject {
         }
     }
     
-    private func delete(_ account: SecureVaultModels.WebsiteAccount) {
+    @discardableResult
+    private func delete(_ account: SecureVaultModels.WebsiteAccount) -> Bool {
         guard let secureVault = try? SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared),
-              let accountID = account.id else { return }
+              let accountID = account.id else { return false }
         
         do {
+            cachedDeletedCredentials = try secureVault.websiteCredentialsFor(accountId: accountID)
             try secureVault.deleteWebsiteCredentialsFor(accountId: accountID)
+            return true
+        } catch {
+            Pixel.fire(pixel: .secureVaultError)
+            return false
+        }
+    }
+    
+    private func undelete(_ account: SecureVaultModels.WebsiteCredentials) {
+        guard let secureVault = try? SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared),
+              var cachedDeletedCredentials = cachedDeletedCredentials else {
+            return
+        }
+        do {
+            // We need to make a new account object. If we try to use the old one, secure vault will try to process it as an update, which will fail
```

Comment:
There are two parts to this PR that I don't love:
1. that it needs to have knowledge of this. Secure vault might change at some point and break assumptions like this
2. That the caching mechanism doesn't live in secure vault. Given how simple it is, I'm not sure it's worth living anywhere else, but would like a second opinion

Reply:
For now, I think this is okay. I'm interesting in adding `UndoManager` support to this UI on macOS, at which point I would take on this work in a project and handle it in a more cross-platform way. For this simple feature, having a bit of logic here in iOS seems acceptable – it's easy to change once we do make some cross-platform friendly changes.

The first concern is definitely a real one... how about adding a unit test for this?

Reply:
I'm happy with that as discussed on MM. Great shout on the unit test (unit tests or lack thereof are why this has the do not merge label atm), that would probably make me happy enough with it